scripts/social-banner/utils.py

```
@@ -32,6 +35,46 @@ def sized_icon(icon, target_size):
+
+def download_icon(url):
+    """Download and decode an icon from *url*, returning a PIL Image.
+
+    Only http/https schemes are allowed. The response is capped at
+    MAX_ICON_BYTES before passing bytes to Image.open.
+    """
+    if urlparse(url).scheme not in ("http", "https"):
+        print(f"Refusing non-http(s) icon URL: {url}", file=sys.stderr)
+        sys.exit(1)
+    try:
+        with urlopen(url, timeout=10) as resp:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)